### PR TITLE
[varlib] add --drop-implied-oncurves option

### PR DESCRIPTION
**Pull request by anthrotype**
Follow up from #3146 

I decided to add this to varLib directly instead of ufo2ft, since the method for dropping the oncurve points was already here.

The option is False by default.
Don't expect much savings, mostly in the hundred of bytes for normal fonts so usually not worth rebuilding yours.

I was thinking of adding a tolerance to increase the chances of pruning oncurves (right now it has to be exactly in the middle, we use math.isclose). But maybe another time, I just need this for testing purposes (diffing against fontc)

### CODE REVIEW

File: Lib/fontTools/ttLib/tables/_g_l_y_f.py [hunk 1]
```diff
@@ -1549,16 +1551,40 @@ def dropImpliedOnCurvePoints(*interpolatable_glyphs: Glyph) -> Set[int]:
...
+            numContours = glyph.numberOfContours
+        elif glyph.numberOfContours != numContours:
+            raise ValueError(
+                f"Incompatible number of contours for glyph at master index {i}: "
+                f"expected {numContours}, found {glyph.numberOfContours}"
+            )
+
+        if flags is None:
+            flags = glyph.flags
+        elif glyph.flags != flags:
```
Comment (behdad) [inline Lib/fontTools/ttLib/tables/_g_l_y_f.py:1580]: Should you also check for same `endPtsOfContours`?

**Review by behdad — APPROVED** [review]
Just a minor comment. LGTM otherwise. Thanks.

File: Lib/fontTools/ttLib/tables/_g_l_y_f.py (hunk 1 above)
Comment (anthrotype) [reply Lib/fontTools/ttLib/tables/_g_l_y_f.py:1580]: Right, it I'd conceivable that numContours and flags be equal but endPtsOfContours are different, will update, thanks

Comment (anthrotype) [reply Lib/fontTools/ttLib/tables/_g_l_y_f.py]: done https://github.com/fonttools/fonttools/pull/3147/commits/5b93100616199945e23cdcd664242b68ad9bf0f8

**Pull request merged**